diff --git a/src/main/java/org/apache/commons/imaging/formats/png/PngImagingParameters.java b/src/main/java/org/apache/commons/imaging/formats/png/PngImagingParameters.java
index af6d4b3a..ba762523 100644
--- a/src/main/java/org/apache/commons/imaging/formats/png/PngImagingParameters.java
+++ b/src/main/java/org/apache/commons/imaging/formats/png/PngImagingParameters.java
@@ -41,6 +41,11 @@ public class PngImagingParameters extends XmpImagingParameters<PngImagingParamet
 
     private boolean predictorEnabled;
 
+    /**
+     * Filter method. Default value is {@literal 0}.
+     */
+    private int filterMethod = 0;
+
     /**
      * Used in write operations to indicate the Physical Scale - sCAL.
      *
@@ -79,6 +84,15 @@ public class PngImagingParameters extends XmpImagingParameters<PngImagingParamet
         return bitDepth;
     }
 
+    /**
+     * Gets the filter method.
+     *
+     * @return the filter method.
+     */
+    public int getFilterMethod() {
+        return filterMethod;
+    }
+
     /**
      * Gets the physical scale.
      *
@@ -135,6 +149,17 @@ public class PngImagingParameters extends XmpImagingParameters<PngImagingParamet
         return asThis();
     }
 
+    /**
+     * Sets the filter method.
+     *
+     * @param filterMethod the filter method.
+     * @return this instance.
+     */
+    public PngImagingParameters setFilterMethod(final int filterMethod) {
+        this.filterMethod = filterMethod;
+        return asThis();
+    }
+
     /**
      * Sets whether to force indexed color.
      *
diff --git a/src/main/java/org/apache/commons/imaging/formats/png/PngWriter.java b/src/main/java/org/apache/commons/imaging/formats/png/PngWriter.java
index c0f1e057..484921ff 100644
--- a/src/main/java/org/apache/commons/imaging/formats/png/PngWriter.java
+++ b/src/main/java/org/apache/commons/imaging/formats/png/PngWriter.java
@@ -362,10 +362,13 @@ public class PngWriter {
             // IHDR must be first
 
             final byte compressionMethod = PngConstants.COMPRESSION_TYPE_INFLATE_DEFLATE;
-            final byte filterMethod = PngConstants.FILTER_METHOD_ADAPTIVE;
+            final int filterMethod = params.getFilterMethod();
+            if (filterMethod < 0 || filterMethod > 4) {
+                throw new ImagingException("PNG: unknown FilterMethod: " + filterMethod);
+            }
             final InterlaceMethod interlaceMethod = InterlaceMethod.NONE;
 
-            final ImageHeader imageHeader = new ImageHeader(width, height, bitDepth, pngColorType, compressionMethod, filterMethod, interlaceMethod);
+            final ImageHeader imageHeader = new ImageHeader(width, height, bitDepth, pngColorType, compressionMethod, (byte) filterMethod, interlaceMethod);
 
             writeChunkIHDR(os, imageHeader);
         }
